### PR TITLE
Some minor fixes to the template, porting over old changes

### PR DESCRIPTION
**Pull request by bhelx**
* Add host functions to index.d.ts
* Allow expressing more than just JSON as host function input/output
* remove propertyIsEmpty helper
* Change version to 0.3.0
* moves xtp-bindgen package to npm  https://github.com/dylibso/xtp-bindgen

### CODE REVIEW

**Comment by bhelx** [comment]
@mhmd-azeez mind giving this a review when you get the chance? more to come but i can follow up next week

File: template/package.json.ejs [hunk 1]
```diff
@@ -9,7 +9,6 @@
   },
   "keywords": [],
   "author": "",
-  "license": "BSD-3-Clause",
```
Comment (bhelx) [inline template/package.json.ejs:12]: perhaps we should also mark these as private. what do you think the defaults of the package.json for a new template should be @chrisdickinson ? this project is of course open source but these plug-ins my be private and commercial.

Comment (chrisdickinson) [reply template/package.json.ejs:12]: We're probably looking for
```
  "license": "UNLICENSED",
  "private": "true",
```

("private" will keep `npm` and friends from ever publishing it!)

Comment (bhelx) [reply template/package.json.ejs:12]: cool, thanks !

Comment (bhelx) [reply template/package.json.ejs:12]: changing now

**Review by mhmd-azeez — APPROVED** [review]
great job! hopefully we will soon be able to remove the go code & output folder too

**Comment by mhmd-azeez** [comment]
I am going to merge this because it causes the tests in xtp to fail (imports are not working properly in the `main` version)

**Pull request merged**